src/context/mod.rs

```diff
@@ -122,9 +124,9 @@ pub trait Context<'a>: ContextInternal<'a> {
     /// Lock the JavaScript engine, returning an RAII guard that keeps the lock active as long as the guard is alive.
     /// 
     /// If this is not the currently active context (for example, if it was used to spawn a scoped context with `execute_scoped` or `compute_scoped`), this method will panic.
-    fn lock(&self) -> Lock {
+    fn lock<'c>(&'c self) -> Lock<'c> {
```

Comment:
Any reason for expanding out the lifetimes here? I prefer using elision when it's available. It makes signatures more lightweight and intuitive.

Reply:
No, is a leftover from an attempt to pass the context reference to the Lock :)
```suggestion
    fn lock(&self) -> Lock<'_> {
```